go/v1/resource_descriptor.go

```diff
@@ -5,29 +5,51 @@ Wrapper APIs for in-toto attestation ResourceDescriptor protos.
 package v1
 
 import (
+	"crypto/md5"
+	"crypto/sha1"
+	"crypto/sha512"
 	"encoding/hex"
 	"errors"
 )
 
 var (
+	ErrIncorrectDigestLength = errors.New("digest is not correct length")
 	ErrInvalidDigestEncoding = errors.New("digest is not valid hex-encoded string")
 	ErrRDRequiredField       = errors.New("at least one of name, URI, or digest are required")
 )
 
+// Supported standard hash algorithms
+func isSupportedAlgorithm(alg string) (bool, int) {
+	algos := map[string]int{"md5": md5.Size, "sha1": sha1.Size, "shake128": md5.Size, "sha224": sha512.Size224, "sha3_224": sha512.Size224, "sha512_224": sha512.Size224, "sha256": sha512.Size256, "sha3_256": sha512.Size256, "sha512_256": sha512.Size256, "shake256": sha512.Size256, "sha384": sha512.Size384, "sha3_384": sha512.Size384, "sha512_384": sha512.Size384, "sha512": sha512.Size, "sha3_512": sha512.Size, "dirHash": sha512.Size256, "gitCommit": sha1.Size}
```

Comment:
should 'shake128' be using md5.Size?  (I don't know much about that hash algorithm).

I'd probably want someone else to take a look at the rest of these too.

BTW can you say a bit about how you came up with these values?

Reply:
I specifically looked up shake128 because I'm also not familiar, and found that it's 16 bytes (size of MD5). But yes, in general, these are based on the hash size constants specified in the respective Go crypto packages. I didn't want to have to import 6 different packages just for the size constants, which is why I use the sha512 constants everywhere possible.

Reply:
Gotcha.  Can you add a comment to that effect?  (Or alternatively use literals)

Reply:
If literals are a bit clearer, I can switch to that.

Reply:
done.